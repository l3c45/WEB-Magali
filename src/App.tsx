import About from "./components/sections/About";
import Contact from "./components/sections/Contact";
import Footer from "./components/sections/Footer";
import Hero from "./components/sections/Hero";
import NavBar from "./components/sections/NavBar";
import Services from "./components/sections/Services";
import useNav from "./hooks/useNav";

const App = () => {
  const [showNav] = useNav();

  return (
    <>
      {showNav ? <NavBar></NavBar> : null}

      <Hero></Hero>
      <About></About>
      <Services></Services>
      <Contact></Contact>
      <Footer></Footer>
    </>
  );
};

export default App;
