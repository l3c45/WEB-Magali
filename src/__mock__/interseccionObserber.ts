// import { vi } from "vitest";


// const intersectionObserverMock = () => ({
//   observe: () => null,
//   disconnect: () => null,

// });
// window.IntersectionObserver = vi.fn().mockImplementation(intersectionObserverMock)
  



import { vi } from 'vitest'

const IntersectionObserverMock = vi.fn(() => ({
  disconnect: vi.fn(),
  observe: vi.fn(),
  takeRecords: vi.fn(),
  unobserve: vi.fn(),
}))

vi.stubGlobal('IntersectionObserver', IntersectionObserverMock)


