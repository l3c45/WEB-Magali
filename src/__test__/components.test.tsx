import { render, screen } from '../utils/testUtils';
import  "../__mock__/interseccionObserber"


import App from '../App';

describe('Testing App component', () => {
  it('renders component', () => {
    render(<App />);
    screen.debug()
  });
});