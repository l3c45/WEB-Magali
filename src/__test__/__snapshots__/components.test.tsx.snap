// Vitest Snapshot v1

exports[`Testing entire App > Should render title  1`] = `
<h1>
   Magali
</h1>
`;
